src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1103,7 +1103,17 @@ public void checkout(Run<?, ?> build, Launcher launcher, FilePath workspace, Tas
 
         // Track whether we're trying to add a duplicate BuildData, now that it's been updated with
         // revision info for this build etc. The default assumption is that it's a duplicate.
-        boolean buildDataAlreadyPresent = build.getActions(BuildData.class).contains(buildData);
+        boolean buildDataAlreadyPresent = false;
+        List<BuildData> actions = build.getActions(BuildData.class);
+        for (BuildData d: actions)  {
+            if (d.similarTo(buildData)) {
+                buildDataAlreadyPresent = true;
+                break;
+            }
+        }
+        if (!actions.isEmpty()) {
+            buildData.setIndex(actions.size()+1);
```

Comment:
Maybe the first action needs to be indexed as well. It increases the risk of the breakage somewhere, but otherwise it may confuse users ("git" and "git-1" in the links)